src/py/flwr/common/supernode_tracker.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""SuperNode Tracker."""
+
+import json
+import os
+from datetime import datetime
+from typing import Any, Dict, List, Optional
+
+from flwr.common import Message
+from flwr.common.typing import Run
+
+
+class SuperNodeTracker:
+    """A utility class for tracking and recording SuperNode."""
+
+    def __init__(self, supernode_id: Optional[int]) -> None:
+        self.supernode_id = str(supernode_id)
+        self.filename = f"supernode_tracking_({supernode_id}).json"
```

Comment:
I might be wrong but appending JSON files into the same file is not a valid JSON file. I think the file extension should be something else.